File: clients/python_pypy/model/player.py
from stream_wrapper import StreamWrapper

class Player:
    """Game's participant (team of units)"""

    __slots__ = ("id","kills","damage","place","score",)

    id: int
    kills: int
    damage: float
    place: int
    score: float

    def __init__(self, id: int, kills: int, damage: float, place: int, score: float):
        self.id = id
        """Unique id"""
        self.kills = kills
        """Number of kills"""
        self.damage = damage
        """Total damage dealt to enemies"""
        self.place = place
        """Survival place (number of survivor teams currently/at the moment of death)"""
        self.score = score
        """Team score"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "Player":
        """Read Player from input stream
        """
        id = stream.read_int()
        kills = stream.read_int()
        damage = stream.read_double()
        place = stream.read_int()
        score = stream.read_double()
        return Player(id, kills, damage, place, score)
    
    def write_to(self, stream: StreamWrapper):
        """Write Player to output stream
        """
        stream.write_int(self.id)
        stream.write_int(self.kills)
        stream.write_double(self.damage)
        stream.write_int(self.place)
        stream.write_double(self.score)
    
    def __repr__(self):
        return "Player(" + \
            repr(self.id) + \
            ", " + \
            repr(self.kills) + \
            ", " + \
            repr(self.damage) + \
            ", " + \
            repr(self.place) + \
            ", " + \
            repr(self.score) + \
            ")"

File: clients/cpp_variant/model/Player.cpp
#include "Player.hpp"

namespace model {

Player::Player(int id, int kills, double damage, int place, double score) : id(id), kills(kills), damage(damage), place(place), score(score) { }

// Read Player from input stream
Player Player::readFrom(InputStream& stream) {
    int id = stream.readInt();
    int kills = stream.readInt();
    double damage = stream.readDouble();
    int place = stream.readInt();
    double score = stream.readDouble();
    return Player(id, kills, damage, place, score);
}

// Write Player to output stream
void Player::writeTo(OutputStream& stream) const {
    stream.write(id);
    stream.write(kills);
    stream.write(damage);
    stream.write(place);
    stream.write(score);
}

// Get string representation of Player
std::string Player::toString() const {
    std::stringstream ss;
    ss << "Player { ";
    ss << "id: ";
    ss << id;
    ss << ", ";
    ss << "kills: ";
    ss << kills;
    ss << ", ";
    ss << "damage: ";
    ss << damage;
    ss << ", ";
    ss << "place: ";
    ss << place;
    ss << ", ";
    ss << "score: ";
    ss << score;
    ss << " }";
    return ss.str();
}

}